xdsl/dialects/dlti.py

```diff
@@ -47,12 +75,25 @@ class DLTIEntryMap(ParametrizedAttribute, ABC):
     # In MLIR, this is a DataLayoutEntryInterface.
     entries: ArrayAttr[DataLayoutEntryAttr]
 
+    def __init__(
+        self,
+        contents: ArrayAttr[DataLayoutEntryAttr] | DictValueType,
+    ):
+        if not isa(contents, ArrayAttr[DataLayoutEntryAttr]):
+            assert isinstance(contents, Mapping)
+            contents = ArrayAttr(
+                [DataLayoutEntryAttr(k, v) for k, v in contents.items()]
+            )
+
+        super().__init__(contents)
+
     @classmethod
     def parse_parameters(
         cls, parser: AttrParser
     ) -> tuple[ArrayAttr[DataLayoutEntryAttr]]:
         def parse_entry() -> DataLayoutEntryAttr:
             entry = parser.parse_attribute()
+            entry = cast(StringAttr | TypeAttribute, entry)
             parser.parse_punctuation("=")
             value = parser.parse_attribute()
             return DataLayoutEntryAttr(entry, value)
```

Comment:
```suggestion
parser.parse_punctuation("=")
value = parser.parse_attribute()
return DataLayoutEntryAttr.new(entry, value)
```

Reply:
Thanks, code updated now to follow this :)